src/insexpand.c

```diff
@@ -4626,35 +4625,26 @@ get_normal_compl_info(char_u *line, int startcol, colnr_T curs_col)
 
     compl_patternlen = STRLEN(compl_pattern);
 
-    if ((get_cot_flags() & COT_FUZZYCOLLECT) != 0)
+    if ((get_cot_flags() & COT_FUZZYCOLLECT) != 0 && compl_length > 0)
     {
-	// Adjust size to avoid buffer overflow
-	fuzzy_len = (size_t)compl_length * 5 + 10;
-	// Allocate enough space
-	fuzzy_pattern = alloc(fuzzy_len);
-	if (fuzzy_pattern == NULL)
-	{
-	    compl_patternlen = 0;
-	    return FAIL;
-	}
-	// Use 'very magic' mode for simpler syntax
-	STRCPY(fuzzy_pattern, "\\v");
+	// Initialize the growable array
+	ga_init2(&fuzzy_ga, sizeof(char), 100);
+	ga_concat(&fuzzy_ga, (char_u *)"\\v\\S*");
 	i = 2; // Start from 2 to skip "\\v"
 	while (i < compl_length + 2)
 	{
-	    // Append "\\k*" before each character
-	    STRNCAT(fuzzy_pattern, "\\k*", fuzzy_len - STRLEN(fuzzy_pattern) - 1);
 	    // Get length of current multi-byte character
 	    char_len = mb_ptr2len(compl_pattern + i);
 	    // Concatenate the character safely
-	    STRNCAT(fuzzy_pattern, compl_pattern + i, char_len);
+	    ga_concat_len(&fuzzy_ga, compl_pattern + i, char_len);
```

Comment:
You can compute the number of bytes to be copied in a tight loop and then use a single `ga_concat_len` to append it, it should be quite faster..